gnucash/gnome/dialog-custom-report.c

```diff
@@ -94,14 +95,40 @@ gboolean custom_report_query_tooltip_cb (GtkTreeView  *view,
                                          GtkTooltip *tooltip,
                                          gpointer    data);
 
+static gboolean
+tree_model_free_guid (GtkTreeModel *model, GtkTreePath *path, GtkTreeIter *iter,
+                      gpointer data)
+{
+    GncGUID *guid;
+    gtk_tree_model_get (model, iter, COL_NUM, &guid, -1);
+    guid_free (guid);
+    return FALSE;
+}
+
+static void
+empty_tree_model (GtkTreeModel *model)
+{
+     gtk_tree_model_foreach (model, (GtkTreeModelForeachFunc)tree_model_free_guid,
+                             NULL);
+     gtk_list_store_clear (GTK_LIST_STORE (model));
+}
+
+void
+custom_report_dialog_destroy_cb (GtkWidget* widget, gpointer data)
+{
+    CustomReportDialog *crd = data;
+    empty_tree_model (gtk_tree_view_get_model (GTK_TREE_VIEW(crd->reportview)));
+    gtk_widget_destroy (crd->dialog);
```

Comment:
I do not think you need this gtk_widget_destroy here, when the close is pressed or 'X' the close callback should call the destroy function which then calls the destroy callback. In that you are then trying to destroy again but crd is freed and so should I think crash. When you 'Alt+F4' which destroys the dialog and as such calls the destroy callback and so again tries to destroy the dialog again and may crash. This may be down to timing and would have to pull your changes and give it a try.